tests/unit/test_kwargs/test_multipart_data.py

```diff
@@ -597,3 +618,100 @@ def test_invalid_multipart_raises_client_error() -> None:
             headers={"Content-Type": "multipart/form-data; charset=utf-8; boundary=20b303e711c4ab8c443184ac833ab00f"},
         )
         assert response.status_code == HTTP_400_BAD_REQUEST
+
+
+# Test for GitHub issue #4204: Empty strings in multipart/form-data should be preserved
+def test_empty_strings_preserved_in_multipart_forms() -> None:
+    """Test that empty strings are preserved in multipart forms and not converted to None.
+
+    This test addresses GitHub issue #4204 where empty strings in multipart/form-data
+    requests were being converted to None instead of being preserved as empty strings.
+    The test focuses on core multipart parsing behavior, not serialization library specifics.
+    """
+
+    @post("/test-form")
+    async def form_handler(request: Request) -> dict[str, Any]:
+        """Handler that directly tests form parsing without any serialization library."""
+        data = await request.form()
+        return {
+            "value": data.get("value"),
+            "value_type": type(data.get("value")).__name__,
+            "value_is_none": data.get("value") is None,
+        }
+
+    with create_test_client([form_handler]) as client:
+        # Test URL-encoded form (baseline behavior)
+        response_url_encoded = client.post("/test-form", data={"value": ""})
+        assert response_url_encoded.status_code == HTTP_201_CREATED
+        url_result = response_url_encoded.json()
+
+        # Test multipart form (should behave identically to URL-encoded)
+        response_multipart = client.post("/test-form", data={"value": ""}, files={"dummy": ""})
+        assert response_multipart.status_code == HTTP_201_CREATED
+        multipart_result = response_multipart.json()
+
+        # Both should preserve empty strings, not convert to None
+        assert url_result["value"] == ""
+        assert multipart_result["value"] == ""
+        assert url_result["value_type"] == "str"
+        assert multipart_result["value_type"] == "str"
+        assert url_result["value_is_none"] is False
+        assert multipart_result["value_is_none"] is False
+
+        # Results should be identical (consistency between form types)
+        assert url_result == multipart_result
```

Comment:
You already have an extra test case below that checks if multipart and url-encoded are the same. There is no need to test the same behaviour multiple times in multiple different tests. See my earlier explanation as to why it's better to keep tests small and focused.